sdk/packages/react/src/hooks/useValidateOrder.ts

```diff
@@ -146,9 +167,12 @@ const isValidateRes = (json: unknown): json is ValidationResponse => {
 }
 
 const useResult = (
+  encoded: EncodeOrderResult,
   query: UseQueryResult<ValidationResponse, FetchJSONError>,
 ): UseValidateOrderResult =>
   useMemo(() => {
+    if (!encoded.ok) return { status: 'error', error: encoded.error }
+
```

Comment:
This pattern makes sense - quicker feedback of encoding errors, and don't need to deal with react-query retrying on errors. So let's keep it 

but wonder it it would work if we just let the async func throw

Reply:
Yes good question, I went with trying to bypass react-query but there might be better ways to handle this

Reply:
react query right now retries on error (4 times) so better to bypass